lib/HALSim/src/Network.cpp

```diff
@@ -155,6 +158,12 @@ bool Network::setConfig(const String& clientId, const String& brokerAddress, uin
     }
     else
     {
+        if (false == birthTopic.isEmpty())

```

Comment:
Is that correct or shouldn't we just assign it? Fact is, you can never clear it after there is something in the m_birthTopic.

Reply:
With this there is the possibility of setting a custom birth/will topic from the application, or none at all. But if this is not necessary, then we can directly assign it.